fog/view/server/tests/router_smoke_tests.rs

```diff
@@ -93,28 +218,210 @@ async fn test_router_integration(test_environment: &mut RouterTestEnvironment, l
     }
 
     let router_client = test_environment.router_client.as_mut().unwrap();
-    let result =
-        mc_fog_view_server_test_utils::assert_e_tx_out_records(router_client, &expected_records)
-            .await;
+    let nonsense_search_keys = vec![vec![50u8]];
+    let result = router_client
+        .query(0, 0, nonsense_search_keys.clone())
+        .await;
+    assert!(result.is_ok());
+    let mut result = result.unwrap();
 
+    assert_eq!(result.highest_processed_block_count, 6);
+    // 4 events are expected (in the following order):
+    // - 1 new rng record (for invoc_id1)
+    // - 1 new rng record (for invoc_id2)
+    // - 1 missing block range
+    // - 1 ingest decommissioning (for invoc_id1)
+    assert_eq!(result.next_start_from_user_event_id, 4);
+    assert_eq!(result.rng_records.len(), 2);
+    result
+        .rng_records
+        .sort_by_key(|rng_record| rng_record.ingest_invocation_id);
+    assert_eq!(result.rng_records[0].pubkey, egress_public_key_1);
+    assert_eq!(result.rng_records[1].pubkey, egress_public_key_2);
+    assert_eq!(result.tx_out_search_results.len(), 1);
+    assert_eq!(
+        TxOutSearchResultCode::try_from(result.tx_out_search_results[0].result_code).unwrap(),
+        TxOutSearchResultCode::BadSearchKey
+    );
+    assert_eq!(result.missed_block_ranges.len(), 1);
+    assert_eq!(result.missed_block_ranges[0], BlockRange::new(3, 4));
+    assert_eq!(result.last_known_block_count, 6);
+
+    let result = router_client
+        .query(0, 0, nonsense_search_keys.clone())
+        .await;
     assert!(result.is_ok());
+    let mut result = result.unwrap();
+    assert_eq!(result.highest_processed_block_count, 6);
+    assert_eq!(result.next_start_from_user_event_id, 4);
+    assert_eq!(result.rng_records.len(), 2);
+    result
+        .rng_records
+        .sort_by_key(|rng_record| rng_record.ingest_invocation_id);
+    assert_eq!(result.rng_records[0].pubkey, egress_public_key_1);
+    assert_eq!(result.rng_records[1].pubkey, egress_public_key_2);
+    assert_eq!(result.tx_out_search_results.len(), 1);
+    assert_eq!(
+        TxOutSearchResultCode::try_from(result.tx_out_search_results[0].result_code).unwrap(),
+        TxOutSearchResultCode::BadSearchKey
+    );
+    assert_eq!(result.missed_block_ranges.len(), 1);
+    assert_eq!(result.missed_block_ranges[0], BlockRange::new(3, 4));
+    assert_eq!(result.last_known_block_count, 6);
 
+    let result = router_client
+        // starting at user event id 2 skips invoc_id1
+        // (event id 1 is for invoc_id1)
+        .query(1, 0, nonsense_search_keys.clone())
```

Comment:
:thought_balloon: This comment is a bit confusing.  It says "user event id 2" then it says "event id 1 is for invoc_id1" and it goes on to use `1` as the `start_from_user_event_id` asserting that only key 2 is present.
Is there an off by one somewhere or is there some wording issues relating to "indices" versus "count"?

Reply:
I updated the comment to be clearer.